### DISCUSSION TITLE
What algorithm is used to detect collision between 2 actors in Carla?

### DISCUSSION THREAD

**Discussion by das007** [carla-simulator/carla#6033] [Q&A]
What algorithm is used to detect collision between 2 actors in Carla?
Can anyone please share the link to the source code?
As per my knowledge, every actor has a collision bounding box. If the collision bounding box for 2 actors overlap, we can say a collision occurred.
But I want to see the source code for this.
Can anyone please help?
Thanks.

**Comment by SailorBrandon** [comment]
Hi,

I am not sure, but this might help 

https://github.com/carla-simulator/carla/blob/a1b37f7f1cf34b0f6f77973c469926ea368d1507/LibCarla/source/carla/trafficmanager/CollisionStage.cpp

**Reply by SailorBrandon** [reply]
This collision detection algorithm looks a bit complicated to me :(  If you figure it out, could you kindly share a bit of detail (such as how the bounding box is selected)?
